### PR TITLE
Remove local database support (and drop MySQL dependency)

### PR DESCRIPTION
**Pull request by lpetre-ulb**
## Description

This PR completely deletes the no longer used local run database. As consequence the MySQL dependency can be dropped as well as the Python dependency (`gemutils` only for the latter).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Clean up.

## How Has This Been Tested?

The `cmsgemos` xDAQ applications still compile, run and their basic actions work.

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by mexanick — DISMISSED** [review]

**Comment by lmoureaux** [comment]
I still see `-I/usr/include/python2.7` all over the place, is it a build system feature?

**Comment by lpetre-ulb** [comment]
> I still see `-I/usr/include/python2.7` all over the place, is it a build system feature?

~~I don't see any in my build tree... Did you start from a clean repo?~~

Oh, in the compilation commands... Good catch! It looks like that every component which used `gemutils` had to add the include directories due to limitations of the build system.

**Comment by lmoureaux** [comment]
I did `make clean; make` and I see them for `gemonlinedb` (sight), `gemreadout`, `gemhardware/managers`, `gemsupervisor`, `gemdaqmonitor` and obviously `gempython`. I grepped for Python includes and couldn't find anything except in `.d` files. Doesn't block this from being merged.

**Comment by lpetre-ulb** [comment]
Useless Python includes removed; as well as outdated `gemutils` DB exceptions.

File: gemsupervisor/xml/gemsupervisor_tif_lemo.xml [hunk 1]
```diff
@@ -138,18 +138,6 @@
...
-          <dbPort xsi:type="xsd:integer">3306</dbPort>
-          <dbUser xsi:type="xsd:string">gemdaq</dbUser>
-          <dbPass xsi:type="xsd:string">gemdaq</dbPass>
-          
-          <setupTag xsi:type="xsd:string">teststand</setupTag>
-          <runPeriod xsi:type="xsd:string">2016T</runPeriod>
-          <setupLocation xsi:type="xsd:string">TIF</setupLocation>
-        </DatabaseInfo>
-
 	<rcmsStateListener xsi:type="soapenc:Struct">
```
Comment (jsturdy) [inline gemsupervisor/xml/gemsupervisor_tif_lemo.xml:141]: To be honest, this file (and all others here not named `gemsupervisor.xml` probably should be removed...

File: gemsupervisor/xml/gemsupervisor_tif.xml [hunk 2]
```diff
@@ -114,18 +114,6 @@
...
-          <dbPort xsi:type="xsd:integer">3306</dbPort>
-          <dbUser xsi:type="xsd:string">gemdaq</dbUser>
-          <dbPass xsi:type="xsd:string">gemdaq</dbPass>
-          
-          <setupTag xsi:type="xsd:string">teststand</setupTag>
-          <runPeriod xsi:type="xsd:string">2016T</runPeriod>
-          <setupLocation xsi:type="xsd:string">TIF</setupLocation>
-        </DatabaseInfo>
-
 	<rcmsStateListener xsi:type="soapenc:Struct">
```
Comment (jsturdy) [inline gemsupervisor/xml/gemsupervisor_tif.xml:117]: can remove

File: gemsupervisor/xml/gemsupervisor_ctp7_stress_test.xml [hunk 3]
```diff
@@ -191,18 +191,6 @@
...
-          <dbHost xsi:type="xsd:string">gem904daq01.cern.ch</dbHost>
-          <dbPort xsi:type="xsd:integer">3306</dbPort>
-          <dbUser xsi:type="xsd:string">gemdaq</dbUser>
-          <dbPass xsi:type="xsd:string">gemdaq</dbPass>
-          
-          <setupTag xsi:type="xsd:string">teststand</setupTag>
-          <runPeriod xsi:type="xsd:string">2016T</runPeriod>
-          <setupLocation xsi:type="xsd:string">CERN904</setupLocation>
-        </DatabaseInfo>
-
```
Comment (jsturdy) [inline gemsupervisor/xml/gemsupervisor_ctp7_stress_test.xml:205]: can remove

File: gemsupervisor/xml/gemsupervisor_tif_lemo.xml (hunk 1 above)
Comment (lpetre-ulb) [reply gemsupervisor/xml/gemsupervisor_tif_lemo.xml:141]: Ok, great! I was hesitating to remove them, but I'll remove them as part of my next cleanup PR (targeting at the removal of unused files).

In the future, the configuration files should probably be stored somewhere else (in the `GEMRunConfigs` repository?) or maintained up-to-date the main repository.

Comment (jsturdy) [reply gemsupervisor/xml/gemsupervisor_tif_lemo.xml:141]: Indeed, *actual* configs should be stored where you mention, but the example configs here should be kept to a minimum and stick with the source (and maybe at some point be automatically generated to stay up-to-date with potential code changes)

**Review by jsturdy — APPROVED** [review]
I'd still remove them here in this PR, since you're already changing them, but non blocking

**Comment by lpetre-ulb** [comment]
Superseded by #324.

**Pull request closed without merging**